ledger/ledger.go

```diff
@@ -321,10 +321,14 @@ func (v Value) MarshalJSON() ([]byte, error) {
 // Migration defines how to convert the given slice of input payloads into an slice of output payloads
 type Migration func(payloads []Payload) ([]Payload, error)
 
+type ExportOutputs struct {
```

Comment:
I'm not sure if I follow, from the separation of concern perspective, Reporter doesn't know anything about an export or a migration, so I would suggest we keep this as part of the specific reporter that needs to export this data instead of requiring it as input for all the reporters. This could be injected as a function to return the previousStateCommitment to the reporter and could be updated with when migration is happening. 

Another alternative if we would like cross reporter and migration data passing, is to update reporter and migration to accept a context.Context object and collect the data as needed.